backend/migrations/20230310_000000_migration.sql

```diff
@@ -0,0 +1,284 @@
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid BIGINT NOT NULL
+, fnname TEXT NOT NULL
+, hash TEXT NOT NULL
+, timestamp TIMESTAMPTZ NOT NULL
+, value TEXT NOT NULL
+, trace_id UUID NOT NULL
+, hash_version INTEGER
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_function_results_v0_most_recent
+ON function_results_v0
+(canvas_id, trace_id, tlid, timestamp DESC);
+
+
+ /* associate it back to the function. */
+CREATE TABLE IF NOT EXISTS packages_v0
+( id UUID PRIMARY KEY
+, tlid BIGINT
+  /* owner/namespace part of the string, eg dark */
+, user_id UUID NOT NULL
+, package TEXT NOT NULL /* eg stdlib */
+, module TEXT NOT NULL /* eg Twitter */
+, fnname TEXT NOT NULL /* eg sendText */
+, version INTEGER NOT NULL /* eg 0 */
+, description TEXT NOT NULL /* docstring */
+, body BYTEA NOT NULL
+, return_type TEXT NOT NULL
+, parameters jsonb NOT NULL
+, author_id UUID NOT NULL /* who uploaded this */
+, deprecated BOOL NOT NULL
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+
+CREATE TYPE scheduling_rule_type AS ENUM ('pause', 'block');
+
+CREATE TABLE IF NOT EXISTS
+scheduling_rules_v0
+( id SERIAL PRIMARY KEY
+, rule_type scheduling_rule_type NOT NULL
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, handler_name TEXT NOT NULL
+, event_space TEXT NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+
+CREATE TABLE IF NOT EXISTS
+secrets_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, secret_name VARCHAR(255) NOT NULL
+, secret_value TEXT NOT NULL
+, secret_version INT NOT NULL DEFAULT 0
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+
+CREATE TABLE IF NOT EXISTS
+stored_events_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, module TEXT NOT NULL
+, path TEXT NOT NULL
+, modifier TEXT NOT NULL
+, timestamp TIMESTAMPTZ NOT NULL
+, value TEXT NOT NULL
+, trace_id UUID NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_stored_events_v0_most_recent
+ON stored_events_v0
+(canvas_id, module, path, modifier, timestamp DESC);
+
+CREATE INDEX IF NOT EXISTS
+stored_events_v0_traceid
+ON stored_events_v0
+(canvas_id, trace_id);
+
+CREATE INDEX IF NOT EXISTS
+idx_stored_events_v0_most_recent_with_text
+ON stored_events_v0
+(canvas_id, module, path text_pattern_ops, modifier, "timestamp" DESC);
+
+
+
+CREATE TABLE IF NOT EXISTS
+system_migrations_v0
+( name TEXT PRIMARY KEY
+, execution_date TIMESTAMPTZ NOT NULL
+, sql TEXT NOT NULL
+);
+
+
+
+CREATE TYPE toplevel_type AS
+ENUM ('handler', 'db', 'user_function');
+
+CREATE TABLE IF NOT EXISTS
+toplevel_oplists_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid INTEGER NOT NULL
+, digest CHAR(32) NOT NULL
+, tipe toplevel_type
+, name TEXT /* handlers only - used for http lookups */
+, module TEXT /* handlers only */
+, modifier TEXT /* handlers only */
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, deleted BOOLEAN
```

Comment:
deleted. oplist, and oplist_cache should be not null

Reply:
Added `NOT NULL` to `deleted`, `oplist`, and `oplist_cache`